packages/app-builder/src/components/Scenario/AstBuilder/AstBuilderNode/Operand/OperandEditor/OperandEditModal/TimestampExtract/TimestampField.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  type AstNode,
+  isTimestampFieldAstNode,
+  type TimestampFieldAstNode,
+} from '@app-builder/models';
+import {
+  useDefaultCoerceToConstant,
+  useGetAstNodeOption,
+  useTimestampFieldOptions,
+} from '@app-builder/services/editor/options';
+import {
+  type AstNodeErrors,
+  type ValidationStatus,
+} from '@app-builder/services/validation/ast-node-validation';
+import * as React from 'react';
+import { toast } from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+
+import { Operand } from '../../../Operand';
+
+export function TimestampField({
+  onChange,
+  astNode,
+  astNodeErrors,
+  validationStatus,
+}: {
+  onChange: (value: TimestampFieldAstNode) => void;
+  astNode: TimestampFieldAstNode;
+  astNodeErrors?: AstNodeErrors;
+  validationStatus: ValidationStatus;
+}) {
+  const { t } = useTranslation(['common', 'scenarios']);
+  const options = useTimestampFieldOptions();
+  const defaultCoerceToConstant = useDefaultCoerceToConstant();
+  const coerceToConstant = React.useCallback(
+    (searchValue: string) =>
+      defaultCoerceToConstant(searchValue).filter(
+        ({ dataType }) => dataType === 'Timestamp',
+      ),
+    [defaultCoerceToConstant],
+  );
+
+  const getAstNodeOption = useGetAstNodeOption();
+
+  const onSave = React.useCallback(
+    (newSelection: AstNode) => {
+      // Safeguard; should never happen and does not protect against every possible case (ex: if the astNode is not a Timestamp data type)
+      // Possible upgrade: add a validation on the operand data type (same condition than the one used inside useTimestampFieldOptions)
+      if (!isTimestampFieldAstNode(newSelection)) {
+        toast.error(t('scenarios:edit_date.field_is_not_supported'));
+        return;
+      }
+      onChange(newSelection);
+    },
+    [onChange, t],
+  );
```

Comment:
I'm not a huge fan of this kind of conditionnal save. By experience, I prefer a structure open to "any kind of modification".
1. Narrowing the possibilities (to prevent errors) can be done using a whitelist of possible values (mostly something like the `coerceToConstant` above already do)
2. In the end, if a "bad" node is saved, this is the validation system that is responsible of informing/helping the user to fix this

IMO, preventing some editting at "save level" is rarelly a good thing.

Still, I must admit that in this perticular case, since we do not support yet "onChange" validation in modals, it could makes sense.

Reply:
Well TBH this file is a pure copy/paste from the file you wrote for the TimeAddEdit modale, as discussed :D 
I have blocking work to finish on my backend PR for Christopher so I'll go ahead and merge this one without reworking the TimestampField component, but point taken and we'll see when we touch this again.

Reply:
NVM, I red to fast.
In that perticular case, since the `onChange` expect a `TimestampFieldAstNode` node, you need to narrow the type (no choice).

This is the only way to do this for now.